sycl/include/sycl/detail/type_traits.hpp

```diff
@@ -532,6 +532,20 @@ struct map_type<T, From, To, Rest...> {
   using type = std::conditional_t<std::is_same_v<From, T>, To,
                                   typename map_type<T, Rest...>::type>;
 };
+template <typename T, typename... Ts> constexpr bool CheckTypeIn() {
+  constexpr bool SameType[] = {
+      std::is_same_v<std::remove_cv_t<T>, std::remove_cv_t<Ts>>...};
+  // Replace with std::any_of with C++20.
```

Comment:
Now that we are looking into this, does it make sense to do what this comment suggests? Maybe in a different PR?

Not a blocking comment, though.

Reply:
We're at C++17, unfortunately...